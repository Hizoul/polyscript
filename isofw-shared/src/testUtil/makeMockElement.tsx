import * as React from "react"

const makeMockElement = (name: string, showChildren: boolean = false) => {
  return (props: any) => (
    <div className={name}>
      {JSON.stringify({...props, children: undefined}, undefined, 2)}
      {props.children}
    </div>
  )
  }

export default makeMockElement
