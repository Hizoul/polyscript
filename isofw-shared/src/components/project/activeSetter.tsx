import { FormStore, IArrayProps, IFieldProps, LoadingStore } from "@xpfw/form-shared"
import { DbStore, IFormShowProps, SharedFormShow } from "@xpfw/ui-shared"
import { IField, prefixMaker } from "@xpfw/validate"
import val from "isofw-shared/src/globals/val"
import { PresetAssistantForm, PresetCameraField, PresetProjectField } from "isofw-shared/src/xpfwDefs/preset"
import { IsActiveField, ProjectCameras, ShotPreset } from "isofw-shared/src/xpfwDefs/project"
import { cloneDeep, get, isBoolean } from "lodash"
import * as React from "react"
import { ComponentBase } from "resub"

const toggleActive = (thisRef: any) => {
  return async (newValue?: any) => {
    await DbStore.getFromServer(thisRef.props.id, val.service.presetAssistant)
  }
}

export interface SharedActiveSetterProps extends IFormShowProps {
  toggleActive: any
  isActive: boolean
}

const SharedActiveSetter = (Container: React.ComponentType<SharedActiveSetterProps>) => {
  class ActiveSetterSub extends React.Component<IFormShowProps, any> {
    private toggleActive: any
    constructor(props: any) {
      super(props)
      this.toggleActive = toggleActive(this)
    }
    public render() {
      return (
        <Container
          {...this.props}
          toggleActive={this.toggleActive}
          isActive={get(this.props.item, IsActiveField.mapTo, false)}
        />
      )
    }
  }
  return SharedFormShow(ActiveSetterSub)
}

export default SharedActiveSetter
export { toggleActive }
