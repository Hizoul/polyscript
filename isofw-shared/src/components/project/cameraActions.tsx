import { cameraCommand } from "isofw-shared/src/cameraApi"
import { DbStore, toJS } from "isofw-shared/src/util/xpfwdata"
import { ExtendedJSONSchema, FormStore, getMapTo, memo } from "isofw-shared/src/util/xpfwform"
import { PresetActionTypeField, PresetAssistantForm, PresetNumberField } from "isofw-shared/src/xpfwDefs/preset"
import { ProjectCameras, ProjectProgram, ProjectShot, ShotCamera, ShotPreset } from "isofw-shared/src/xpfwDefs/project"
import { get } from "lodash"
import usePresetUpdater from "../preset/updater"

const actionPrefix = "actionButtons"

const makeCameraAction = (id: string, patchData: any, prefix?: string) => {
  return async () => {
    FormStore.setValue(String(PresetAssistantForm.title), patchData, prefix)
    const updateRes = await DbStore.patch(id, PresetAssistantForm, undefined, prefix)
    return updateRes
  }
}

const useCameraActions = (schema: ExtendedJSONSchema, mapTo?: string, prefix?: string) => {
  let cameras = FormStore.getValue(ProjectCameras.title, prefix)
  cameras = cameras ? cameras : []
  const value = FormStore.getValue(ShotCamera.title, prefix, cameras[0])
  const currentPreset = FormStore.getValue(ProjectProgram.title, prefix, {})[FormStore.getValue(ProjectShot.title, prefix, 0) - 1]
  const presetHelper = usePresetUpdater(get(currentPreset, ShotPreset.title))
  return {
    cameras,
    value,
    currentPreset,
    loading: false,
    ...presetHelper,
    stopZoom: memo(() => makeCameraAction(value, {
      [String(PresetActionTypeField.title)]: cameraCommand.doZoom,
      [String(PresetNumberField.title)]: 50
    }, prefix), ["zoom", prefix, value, 50]),
    wideZoom: memo(() => makeCameraAction(value, {
      [String(PresetActionTypeField.title)]: cameraCommand.doZoom,
      [String(PresetNumberField.title)]: 55
    }, prefix), ["zoom", prefix, value, 55]),
    teleZoom: memo(() => makeCameraAction(value, {
      [String(PresetActionTypeField.title)]: cameraCommand.doZoom,
      [String(PresetNumberField.title)]: 45
    }, prefix), ["zoom", prefix, value, 45])
  }
}

export default useCameraActions
