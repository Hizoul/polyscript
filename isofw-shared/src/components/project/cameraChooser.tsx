import { DbStore } from "isofw-shared/src/util/xpfwdata"
import { ExtendedJSONSchema, FormStore, getMapTo, memo, prependPrefix } from "isofw-shared/src/util/xpfwform"
import { PresetAssistantForm, PresetCameraField, PresetProjectField } from "isofw-shared/src/xpfwDefs/preset"
import { ProjectCameras, ShotPreset } from "isofw-shared/src/xpfwDefs/project"
import { flow } from "mobx"

const popupVisibilityKey = "cameraChoice"
const freePresetKey = "freePresetGetter"

const togglePop = (schema: ExtendedJSONSchema, mapTo?: string, prefix?: string, newValue?: any) => {
  if (mapTo == null) { mapTo = getMapTo(schema, mapTo)}
  return () => {
    FormStore.setValue(mapTo, newValue, prependPrefix(prefix, popupVisibilityKey))
  }
}
const untypedDbStore: any = DbStore

const setValueWithPreset = (schema: ExtendedJSONSchema, mapTo?: any, prefix?: string) => {
  if (mapTo == null) { mapTo = getMapTo(schema, mapTo)}
  return flow(function *(newValue?: any) {
    const creationPrefix = prependPrefix(`${prependPrefix(PresetAssistantForm.title, freePresetKey)}`, prefix)
    FormStore.setValue(getMapTo(schema, mapTo), newValue, prefix)
    FormStore.setValue(PresetAssistantForm.title, {}, prependPrefix(freePresetKey, prefix))
    FormStore.setValue(PresetCameraField.title, newValue, creationPrefix)
    FormStore.setValue(PresetProjectField.title, untypedDbStore.currentlyEditing, creationPrefix)
    const freeId: any = yield DbStore.create(PresetAssistantForm, undefined, prependPrefix(freePresetKey, prefix))
    mapTo = mapTo.substring(0, mapTo.indexOf("]") + 1)
    FormStore.setValue(`${prependPrefix(mapTo, prefix)}.${ShotPreset.title}`, freeId)
    return freeId
  })
}

const emptyArray = (size: number) => {
  const arr: any[] = []
  for (let index = 0; index < size; index++) {
    arr.push(undefined)
  }
  return arr
}

const useCameraChooser = (schema: ExtendedJSONSchema, mapTo?: string, prefix?: string) => {
  if (mapTo == null) { mapTo = getMapTo(schema, mapTo) }
  let cameras = FormStore.getValue(ProjectCameras.title, prefix)
  cameras = cameras ? cameras : []
  const arr = mapTo.substring(0, mapTo.indexOf("["))
  const pop = FormStore.getValue(arr, prependPrefix(prefix, popupVisibilityKey))
  if (pop == null || pop.length < 600) {
    FormStore.setValue(arr, emptyArray(602), prependPrefix(prefix, popupVisibilityKey))
  }
  return {
    cameras,
    value: FormStore.getValue(getMapTo(schema, mapTo), prefix),
    showPopUp: FormStore.getValue(getMapTo(schema, mapTo), prependPrefix(prefix, popupVisibilityKey), false) === true,
    hidePop: memo(() => togglePop(schema, mapTo, prefix, false), ["hidePop", mapTo, prefix, JSON.stringify(schema)]),
    showPop: memo(() => togglePop(schema, mapTo, prefix, true), ["showPop", mapTo, prefix, JSON.stringify(schema)]),
    setValueWithPreset: memo(() => setValueWithPreset(schema, mapTo, prefix), ["setValueWithPreset", JSON.stringify(schema), mapTo, prefix])
  }
}

export default useCameraChooser
export { togglePop, setValueWithPreset }
