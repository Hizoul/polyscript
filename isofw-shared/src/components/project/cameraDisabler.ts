import {
  ExtendedJSONSchema, FormStore, getMapTo, IFieldProps,
  memo, prependPrefix, useFieldWithValidation
} from "isofw-shared/src/util/xpfwform"
import { OperatorRelation, ProjectCameras, ProjectOperatorCameraMapping, ProjectOperators } from "isofw-shared/src/xpfwDefs/project"
import { cloneDeep, findIndex, get } from "lodash"
import { action } from "mobx"

const toggleCamera = (schema: ExtendedJSONSchema, mapTo?: string, prefix?: string, camera?: string) => {
  return action((event?: any) => {
    const fieldHelper = useFieldWithValidation(schema, mapTo, prefix)
    let currentArray = fieldHelper.value
    if (currentArray == null) {
      currentArray = []
    }
    const cameraIndex = currentArray.indexOf(camera)
    if (cameraIndex === -1) {
      currentArray.push(camera)
    } else {
      currentArray.splice(cameraIndex, 1)
    }
    fieldHelper.setValue(currentArray)
  })
}
const toggleOperator = (schema: ExtendedJSONSchema, mapTo?: string, prefix?: string, operator?: string) => {
  return action((event?: any) => {
    const fieldHelper = useFieldWithValidation(schema, mapTo, prefix)
    let currentArray = fieldHelper.value
    if (currentArray == null) {
      currentArray = []
    }
    const operatorCameraMapping = FormStore.getValue(String(ProjectOperatorCameraMapping.title), prefix, [])
    for (const entry of operatorCameraMapping) {
      const operatorId = entry[String(OperatorRelation.title)]
      if (operatorId === operator) {
        let hasActiveCamera = false
        for (const cameraId of entry[String(ProjectCameras.title)]) {
          if (currentArray.indexOf(cameraId) === -1) {
            hasActiveCamera = true
          }
        }
        for (const cameraId of entry[String(ProjectCameras.title)]) {
          const cameraIndex = currentArray.indexOf(cameraId)
          if (hasActiveCamera && cameraIndex === -1) {
            currentArray.push(cameraId)
          } else if (!hasActiveCamera && cameraIndex !== -1) {
            currentArray.splice(cameraIndex, 1)
          }
        }
      }
    }
  })
}

export interface CameraDisablerUtils {
  value: any
  operatorCameraMapping: any[]
  cameraToggles: any
  operatorToggles: any
  isActive: any
}

const useCameraDisabler = (schema: ExtendedJSONSchema, mapTo?: string, prefix?: string) => {
  if (mapTo == null) { mapTo = getMapTo(schema, mapTo) }
  const field = useFieldWithValidation(schema, mapTo, prefix)
  let value = field.value
  if (value == null) {
    value = []
  }
  const operatorCameraMapping = FormStore.getValue(String(ProjectOperatorCameraMapping.title), prefix, [])
  const cameraToggles: any = {}
  const operatorToggles: any = {}
  const isActive: any = {}
  for (const entry of operatorCameraMapping) {
    const operatorId = entry[String(OperatorRelation.title)]
    operatorToggles[operatorId] = memo(() => toggleOperator(schema, mapTo, prefix, operatorId),
      ["toggleOperator", mapTo, prefix, operatorId])
    let hasActiveCamera = false
    for (const cameraId of entry[String(ProjectCameras.title)]) {
      cameraToggles[cameraId] = memo(() => toggleCamera(schema, mapTo, prefix, cameraId),
        ["toggleCamera", mapTo, prefix, operatorId])
      isActive[cameraId] = value.indexOf(cameraId) === -1
      if (isActive[cameraId] === true) {
        hasActiveCamera = true
      }
    }
    isActive[operatorId] = hasActiveCamera
  }
  const ret: CameraDisablerUtils = {
    value,
    operatorCameraMapping,
    isActive,
    cameraToggles,
    operatorToggles
  }
  return ret
}

export default useCameraDisabler
export {
  toggleCamera, toggleOperator
}
