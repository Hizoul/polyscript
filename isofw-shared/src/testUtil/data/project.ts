import { MailField, PwField } from "@xpfw/ui-shared"
import val from "isofw-shared/src/globals/val"
import isServerParams from "isofw-shared/src/globals/isServerParams"
import { ProjectForm, ProjectName, ProjectShot } from "isofw-shared/src/xpfwDefs/project";

const testProjects = [
  {_id: "b19999999999999999999999", [ProjectName.mapTo]: "pn1", [ProjectShot.mapTo]: 1},
  {_id: "b29999999999999999999999", [ProjectName.mapTo]: "pn2", [ProjectShot.mapTo]: 8},
  {_id: "b39999999999999999999999", [ProjectName.mapTo]: "pn3", [ProjectShot.mapTo]: 6}
]
8
const createTestProjects = async (app: any) => {
  const newProjects = []
  for (const newUser of testProjects) {
    newProjects.push(await app.service(val.service.project).create(newUser, isServerParams))    
  }
  return newProjects
}

export default createTestProjects
export {
  testProjects
}