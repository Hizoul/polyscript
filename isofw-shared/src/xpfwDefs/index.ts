import { dataOptions } from "@xpfw/data"
dataOptions.idPath = "_id"

import "./camera"
import "./preset"
import "./project"
import "./user"
