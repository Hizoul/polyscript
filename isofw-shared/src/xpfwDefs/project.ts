import { MailField } from "@xpfw/data"
import { addTimeStamp, ExtendedJSONSchema, executeForMethods } from "@xpfw/form"
import val from "isofw-shared/src/globals/val"
import { cloneDeep } from "lodash"
import { IDField } from "./commonFields"
import { changeValToRegex } from "@xpfw/data"
import { changeValToRegexNedb } from "../util/valToRegex"

const ProjectName: ExtendedJSONSchema = {
  type: "string",
  title: "name"
}

const ProjectShot: ExtendedJSONSchema = {
  type: "number",
  title: "shot"
}

const InstrumentForm: ExtendedJSONSchema = {
  title: "projectModel",
  collection: val.service.instruments,
  type: "object",
  properties: {
    [String(IDField.title)]: IDField,
    [String(ProjectName.title)]: ProjectName
  }
}

const ShotName: ExtendedJSONSchema = {
  type: "string",
  title: "pname"
}

const ShotMovement: ExtendedJSONSchema = {
  type: "string",
  title: "movement"
}

const ShotMovementTowards: ExtendedJSONSchema = {
  type: "string",
  title: "movementTowards"
}

const ShotDuration: ExtendedJSONSchema = {
  type: "number",
  title: "duration"
}

const ShotType: ExtendedJSONSchema = {
  type: "string",
  title: "type"
}

const ShotRemarksDirector: ExtendedJSONSchema = {
  type: "string",
  title: "directorRemarks"
}

const ShotRemarksOperator: ExtendedJSONSchema = {
  type: "string",
  title: "operatorRemarks"
}

const ShotCamera: ExtendedJSONSchema = {
  type: "string",
  title: "camera",
  theme: "program",
  relationship: {
    namePath: ProjectName.title,
    collection: "cameras",
    idPath: "_id"
  }
}

const ShotPreset: ExtendedJSONSchema = {
  type: "string",
  title: "preset",
  theme: "presetNumberDisplay"
}

const ShotNumber: ExtendedJSONSchema = {
  type: "number",
  title: "number"
}

const ShotImportance: ExtendedJSONSchema = {
  type: "string",
  title: "importance",
  theme: "select",
  selectOptions: [
    {label: "Normal", value: "n"},
    {label: "Raised", value: "r"},
    {label: "Maximum", value: "m"}
  ]
}

const ProjectProgram: ExtendedJSONSchema = {
  type: "array",
  title: "program",
  items: {
    type: "object",
    properties: {
      [String(ShotName.title)]: ShotName,
      [String(ShotType.title)]: ShotType,
      [String(ShotMovement.title)]: ShotMovement,
      [String(ShotMovementTowards.title)]: ShotMovementTowards,
      [String(ShotDuration.title)]: ShotDuration,
      [String(ShotRemarksDirector.title)]: ShotRemarksDirector,
      [String(ShotRemarksOperator.title)]: ShotRemarksOperator,
      [String(ShotCamera.title)]: ShotCamera,
      [String(ShotPreset.title)]: ShotPreset,
      [String(ShotImportance.title)]: ShotImportance,
      [String(ShotNumber.title)]: ShotNumber
    }
  },
  hide: {update: true, create: true}
}

const ProjectCameras: ExtendedJSONSchema = {
  type: "array",
  title: "cameras",
  theme: "multi",
  relationship: {
    namePath: ProjectName.title,
    collection: "cameras",
    idPath: "_id"
  },
  items: {type: "string"}
}

const DisabledCameras: ExtendedJSONSchema = cloneDeep(ProjectCameras)
DisabledCameras.title = "disabledCameras"
DisabledCameras.theme = "cameraDisabler"

const ProjectOperators: ExtendedJSONSchema = {
  type: "array",
  title: "operators",
  items: {type: "string"},
  theme: "multi",
  relationship: {
    namePath: MailField.title,
    collection: "users",
    idPath: "_id"
  }
}

const OperatorRelation: ExtendedJSONSchema = {
  type: "string",
  title: "operator",
  theme: "single",
  relationship: {
    namePath: MailField.title,
    collection: "users",
    idPath: "_id"
  }
}

const ProjectOperatorCameraMapping: ExtendedJSONSchema = {
  type: "array",
  title: "operatorCameraMapping",
  theme: "cameraMapping",
  items: {
    type: "object",
    properties: {
      [String(OperatorRelation.title)]: OperatorRelation,
      [String(ProjectCameras.title)]: ProjectCameras
    }
  }
}

const IsActiveField: ExtendedJSONSchema = {
  type: "boolean",
  title: "isActive",
  default: true,
  hide: {create: false, update: false}
}

const ProjectForm: ExtendedJSONSchema = {
  title: "projectModel",
  collection: val.service.project,
  type: "object",
  properties: {
    [String(IDField.title)]: IDField,
    [String(ProjectName.title)]: ProjectName,
    [String(ProjectShot.title)]: ProjectShot,
    [String(ProjectCameras.title)]: ProjectCameras,
    [String(ProjectOperators.title)]: ProjectOperators,
    [String(ProjectOperatorCameraMapping.title)]: ProjectOperatorCameraMapping,
    [String(IsActiveField.title)]: IsActiveField,
    [String(DisabledCameras.title)]: DisabledCameras
  },
  modify: [addTimeStamp("createdAt", ["create"]), (val.useNedb ? changeValToRegexNedb(String(ProjectName.title), ["find"]) : changeValToRegex(String(ProjectName.title), ["find"]))]
}

const StrippedProjectForm: ExtendedJSONSchema = {
  title: "projectModel",
  collection: val.service.project,
  type: "object",
  properties: {
    [String(ProjectName.title)]: ProjectName,
    [String(ProjectShot.title)]: ProjectShot,
    [String(ProjectCameras.title)]: ProjectCameras,
    [String(ProjectOperators.title)]: ProjectOperators,
    [String(ProjectOperatorCameraMapping.title)]: ProjectOperatorCameraMapping
  },
  modify: [addTimeStamp("createdAt", ["create"])]
}

export {
  ProjectForm, ProjectName, ProjectShot, ProjectProgram, ShotCamera, ShotPreset, ShotImportance,
  ProjectOperators, ProjectOperatorCameraMapping, OperatorRelation, ProjectCameras, IsActiveField,
  ShotName, ShotType, ShotMovement, ShotMovementTowards, ShotDuration, ShotRemarksDirector, ShotRemarksOperator,
  DisabledCameras, ShotNumber, StrippedProjectForm, InstrumentForm
}
