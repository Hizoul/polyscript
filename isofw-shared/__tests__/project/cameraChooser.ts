import { ComponentRegistry } from "@xpfw/form-shared"
import { FieldType } from "@xpfw/validate"
import useCameraChooser from "isofw-shared/src/components/project/cameraChooser"
import cameraChooserTest from "isofw-shared/src/tests/project/cameraChooser"
import makeMockElement from "isofw-shared/src/testUtil/makeMockElement"
import { ShotCamera } from "isofw-shared/src/xpfwDefs/project"

ComponentRegistry.registerComponent(FieldType.RelationshipSingle,
  makeMockElement("DirectorEle", (props) => useCameraChooser(props.schema, props.mapTo, props.prefix)),
  ShotCamera.theme)
cameraChooserTest()
