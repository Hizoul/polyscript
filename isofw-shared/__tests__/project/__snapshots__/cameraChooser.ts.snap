// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` camera mapping test  works as planned:  after Closing pop up  1`] = `
<div
  className="DirectorEle"
>
  {
  "cameras": [
    "c1bbbbbbbbbbbbbbbbbbbbbb",
    "c2bbbbbbbbbbbbbbbbbbbbbb",
    "c3bbbbbbbbbbbbbbbbbbbbbb"
  ],
  "operators": {
    "c1bbbbbbbbbbbbbbbbbbbbbb": [],
    "c2bbbbbbbbbbbbbbbbbbbbbb": [],
    "c3bbbbbbbbbbbbbbbbbbbbbb": []
  },
  "value": "c2bbbbbbbbbbbbbbbbbbbbbb",
  "showPopUp": false
}
</div>
`;

exports[` camera mapping test  works as planned:  after Fetching cameras  1`] = `
<div
  className="DirectorEle"
>
  {
  "cameras": [
    "c1bbbbbbbbbbbbbbbbbbbbbb",
    "c2bbbbbbbbbbbbbbbbbbbbbb",
    "c3bbbbbbbbbbbbbbbbbbbbbb"
  ],
  "operators": {
    "c1bbbbbbbbbbbbbbbbbbbbbb": [],
    "c2bbbbbbbbbbbbbbbbbbbbbb": [],
    "c3bbbbbbbbbbbbbbbbbbbbbb": []
  },
  "value": "c2bbbbbbbbbbbbbbbbbbbbbb",
  "showPopUp": false
}
</div>
`;

exports[` camera mapping test  works as planned:  after Opening pop up  1`] = `
<div
  className="DirectorEle"
>
  {
  "cameras": [
    "c1bbbbbbbbbbbbbbbbbbbbbb",
    "c2bbbbbbbbbbbbbbbbbbbbbb",
    "c3bbbbbbbbbbbbbbbbbbbbbb"
  ],
  "operators": {
    "c1bbbbbbbbbbbbbbbbbbbbbb": [],
    "c2bbbbbbbbbbbbbbbbbbbbbb": [],
    "c3bbbbbbbbbbbbbbbbbbbbbb": []
  },
  "value": "c2bbbbbbbbbbbbbbbbbbbbbb",
  "showPopUp": true
}
</div>
`;

exports[` camera mapping test  works as planned:  after Setting Camera  1`] = `
<div
  className="DirectorEle"
>
  {
  "cameras": [
    "c1bbbbbbbbbbbbbbbbbbbbbb",
    "c2bbbbbbbbbbbbbbbbbbbbbb",
    "c3bbbbbbbbbbbbbbbbbbbbbb"
  ],
  "operators": {
    "c1bbbbbbbbbbbbbbbbbbbbbb": [],
    "c2bbbbbbbbbbbbbbbbbbbbbb": [],
    "c3bbbbbbbbbbbbbbbbbbbbbb": []
  },
  "value": "c2bbbbbbbbbbbbbbbbbbbbbb",
  "showPopUp": false
}
</div>
`;

exports[` camera mapping test  works as planned:  after preparing cameras  1`] = `
<div
  className="DirectorEle"
>
  {
  "cameras": [
    "c1bbbbbbbbbbbbbbbbbbbbbb",
    "c2bbbbbbbbbbbbbbbbbbbbbb",
    "c3bbbbbbbbbbbbbbbbbbbbbb"
  ],
  "operators": {
    "c1bbbbbbbbbbbbbbbbbbbbbb": [],
    "c2bbbbbbbbbbbbbbbbbbbbbb": [],
    "c3bbbbbbbbbbbbbbbbbbbbbb": []
  },
  "showPopUp": false
}
</div>
`;

exports[` camera mapping test  works as planned: Before anything 1`] = `
<div
  className="DirectorEle"
>
  {
  "cameras": [],
  "operators": {},
  "showPopUp": false
}
</div>
`;
